src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+
+describe('ReactPropTypesProduction', function() {
+  var oldProcess;
+  var PropTypes;
+  var React;
+  var ReactTestUtils;
+
+  beforeEach(function() {
+    __DEV__ = false;
+    oldProcess = process;
+    global.process = {env: {NODE_ENV: 'production'}};
+
+    jest.resetModuleRegistry();
+    PropTypes = require('ReactPropTypes');
+    React = require('React');
+    ReactTestUtils = require('ReactTestUtils');
+  });
+
+  afterEach(function() {
+    __DEV__ = true;
+    global.process = oldProcess;
+  });
+
+  function typeCheckPass(declaration, value) {
+    var props = {testProp: value};
+    declaration(
+      props,
+      'testProp',
+      'testComponent',
+      'prop'
+    );
+    const expectedCount = __DEV__ ? 0 : 1;
+    expect(console.error.calls.count()).toBe(expectedCount);
+    if (!__DEV__) {
+      expect(console.error.calls.argsFor(0)[0]).toContain(
+        'You are manually calling a React.PropTypes validation '
+      );
+    }
+    console.error.calls.reset();
+  }
+
+
+  describe('Primitive Types', function() {
+    it('should warn if called manually in production', function() {
+      spyOn(console, 'error');
+      typeCheckPass(PropTypes.array, /please/);
```

Comment:
Maybe I should rename `typeCheckPass`. It doesn't actually test if the validation passes or fails since that's tested elsewhere. I used valid and invalid values to be sure that the error is logged regardless of whether the validation passes or not.